core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -210,7 +212,7 @@ public WriteResult(T response, Translog.Location location) {
 
     class OperationTransportHandler implements TransportRequestHandler<Request> {
         @Override
-        public void messageReceived(final Request request, final TransportChannel channel) throws Exception {
+        public void messageReceived(final Request request, final TransportChannel channel, Task task) throws Exception {
```

Comment:
same suggestion  as for master operation use java8 default impl in _TransportRequestHandler_:

```Java
default void messageReceived(final Request request, final TransportChannel channel, Task task) {
  messageReceived(request, channel); // override if you need the task
}
````